### ISSUE TITLE
CPE comparison failure in Windows vulnerability detector

### ISSUE THREAD

**Issue by DFolchA** [wazuh/wazuh#5161]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.13.0 | vulnerability detector | Manager | Packages/Sources | Linux |

Hello team,

We found that when the CPE obtained from the NVD database contains an architecture, the comparison with the package CPE calculated from the `cpe_helper.json` fails.

How to reproduce:

Import an NVD feed from a file with this NVD feed configuration:
```
   <provider name="nvd">
      <enabled>yes</enabled>
      <path>/path/to/feed.json</path>
      <update_from_year>2019</update_from_year>
      <update_interval>1h</update_interval>
    </provider>
```
In the custom feed add a package with an entry like this where the `cpe23Uri` contains an architecture:

```
{
    "cve" : {
      "data_type" : "CVE",
      "data_format" : "MITRE",
      "data_version" : "4.0",
      "CVE_data_meta" : {
        "ID" : "CVE-000",
        "ASSIGNER" : "WAZUH"
      },
      "affects" : {
        "vendor" : {
          "vendor_data" : [ {
            "vendor_name" : "wazuhintegrationtests",
            "product" : {
              "product_data" : [ {
                "product_name" : "wazuhintegrationpackage-0",
                "version" : {
                  "version_data" : [ {
                    "version_value" : "1.0.0",
                    "version_affected" : "="
                  } ]
                }
              } ]
            }
          } ]
        }
      },
      "problemtype" : {
        "problemtype_data" : [ {
          "description" : [ {
            "lang" : "en",
            "value" : "CWE-200"
          } ]
        } ]
      },
      "references" : {
        "reference_data" : [ {
          "url" : "https://github.com/wazuh/wazuh-qa/",
          "name" : "WAZUH-INTEGRATION-WVE-000",
          "refsource" : "WAZUH",
          "tags" : [ ]
        }]
      },
      "description" : {
        "description_data" : [ {
          "lang" : "en",
          "value" : "Wazuh integration test NVD vulnerability"
        } ]
      }
    },
    "configurations" : {
      "CVE_data_version" : "4.0",
      "nodes" : [ {
        "operator" : "OR",
        "cpe_match" : [ {
          "vulnerable" : true,
          "cpe23Uri" : "cpe:2.3:a:wazuhintegrationtests:wazuhintegrationpackage-0:*:*:*:*:*:*:x86_64:*",
          "versionEndExcluding" : "2.0.0"
        } ]
      } ]
    },
    "impact" : {
      "baseMetricV2" : {
        "cvssV2" : {
          "version" : "2.0",
          "vectorString" : "AV:N/AC:L/Au:N/C:C/I:C/A:C",
          "accessVector" : "LOCAL",
          "accessComplexity" : "LOW",
          "authentication" : "NONE",
          "confidentialityImpact" : "PARTIAL",
          "integrityImpact" : "NONE",
          "availabilityImpact" : "NONE",
          "baseScore" : 10
        },
        "severity" : "CRITICAL",
        "exploitabilityScore" : 10,
        "impactScore" : 10,
        "acInsufInfo" : false,
        "obtainAllPrivilege" : false,
        "obtainUserPrivilege" : false,
        "obtainOtherPrivilege" : false,
        "userInteractionRequired" : false
      }
    },
    "publishedDate" : "2019-11-19T22:15Z",
    "lastModifiedDate" : "2020-01-21T01:15Z"
  }
```
Then add the corresponding package to the package database of a windows agent, and the corresponding entry in cpe_helper.json:

```
       {
            "target": "windows",
            "source": {
                "vendor": [
                    "^WazuhIntegrationTests*"
                ],
                "product": [
                    "^wazuhintegrationpackage*"
                ]
            },
            "translation": {
                "vendor": [
                    "wazuhintegrationtests"
                ],
                "product": [
                    "wazuhintegrationpackage-1"
                ],
                "version": []
            },
            "action": [
                "replace_vendor",
                "replace_product"
            ]
        }
```

Then the alerts that should appear will not appear.

Regards,
Daniel Folch

**Comment by JcabreraC** [comment]
The problem is occurring in the function `wm_vuldet_get_vuln_nvd_cpe`, in the following if :
```c
if (wm_vuldet_prepare(db, vu_queries[VU_GET_NVD_CPE], -1, &stmt, NULL) != SQLITE_OK) {
        return wm_vuldet_sql_error(db, stmt);
    }
    sqlite3_bind_text(stmt, 1, agent_cpe->part, -1, NULL);
    sqlite3_bind_text(stmt, 2, agent_cpe->vendor, -1, NULL);
    sqlite3_bind_text(stmt, 3, agent_cpe->product, -1, NULL);
    sqlite3_bind_text(stmt, 4, agent_cpe->version, -1, NULL);
    sqlite3_bind_text(stmt, 5, agent_cpe->update ? agent_cpe->update : "", -1, NULL);
    sqlite3_bind_text(stmt, 6, agent_cpe->sw_edition ? agent_cpe->sw_edition : "", -1, NULL);
    sqlite3_bind_text(stmt, 7, agent_cpe->target_hw ? agent_cpe->target_hw : "", -1, NULL);
    while(result = wm_vuldet_step(stmt), result != SQLITE_DONE) {
        .
        .
    }
   .
   .
}
```
If we're looking for the query:
```sqlite
SELECT NVD_CPE.ID FROM NVD_CPE WHERE NVD_CPE.PART = ? AND NVD_CPE.VENDOR = ? AND NVD_CPE.PRODUCT = ? AND (NVD_CPE.VERSION = ? OR NVD_CPE.VERSION = '*' OR NVD_CPE.VERSION = '-') AND (NVD_CPE.UPDATED = '*' OR NVD_CPE.UPDATED = ?) AND (NVD_CPE.EDITION = '*' OR NVD_CPE. EDITION = ? OR NVD_CPE.EDITION = '') AND (NVD_CPE.LANGUAGE = '*' OR NVD_CPE.LANGUAGE = ? OR NVD_CPE.LANGUAGE = '') AND (NVD_CPE.SW_EDITION = '*' OR NVD_CPE.SW_EDITION = ? OR NVD_CPE.SW_EDITION = '') AND (NVD_CPE.TARGET_SW = '*' OR NVD_CPE.TARGET_SW = '-' OR NVD_CPE.TARGET_SW = '') AND (NVD_CPE.TARGET_HW = '*' OR NVD_CPE.TARGET_HW = '-' OR NVD_CPE.TARGET_HW = ?);
```

We see that the number of parameters is incorrect. Because of this, we insert the architecture in a field that is not correct. When calling the function `wm_vuldet_step(stmt)`, it returns SQLITE_DONE, skipping all checks.

We only inserted seven of the nine fields. In addition, we entered two fields in incorrect positions
The following fields of the query:
```c
sqlite3_bind_text(stmt, 6, agent_cpe->sw_edition ? agent_cpe->sw_edition : "", -1, NULL);
sqlite3_bind_text(stmt, 7, agent_cpe->target_hw ? agent_cpe->target_hw : "", -1, NULL);
```
should be:
```c
sqlite3_bind_text(stmt, 8, agent_cpe->sw_edition ? agent_cpe->sw_edition : "", -1, NULL);
sqlite3_bind_text(stmt, 9, agent_cpe->target_hw ? agent_cpe->target_hw : "", -1, NULL);
```

On the other hand, we should complete the `NVD_CPE. EDITION` and `NVD_CPE.LANGUAGE` field:
```c
sqlite3_bind_text(stmt, 6, agent_cpe->edition ? agent_cpe->edition : "", -1, NULL);
sqlite3_bind_text(stmt, 7, agent_cpe->language ? agent_cpe->language : "", -1, NULL);
```

**Comment by chemamartinez** [comment]
Fixed at #5168. Thanks, @JcabreraC and @DFolchA.
